From git@z Thu Jan  1 00:00:00 1970
Subject: [PATCH] backlight: lp855x: Drop ret variable in brightness change
 function
From: Artur Weber <aweber.kernel@gmail.com>
Date: Wed, 09 Aug 2023 13:42:16 +0200
Message-Id: <20230809114216.4078-1-aweber.kernel@gmail.com>
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 7bit

Fixes the following warning:

drivers/video/backlight/lp855x_bl.c:252:7: warning: variable 'ret' is used
uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]

Signed-off-by: Artur Weber <aweber.kernel@gmail.com>
Fixes: 5145531be5fb ("backlight: lp855x: Catch errors when changing brightness")
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202308091728.NEJhgUPP-lkp@intel.com/
Reviewed-by: Daniel Thompson <daniel.thompson@linaro.org>
Link: https://lore.kernel.org/r/20230809114216.4078-1-aweber.kernel@gmail.com
---
 drivers/video/backlight/lp855x_bl.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/video/backlight/lp855x_bl.c b/drivers/video/backlight/lp855x_bl.c
index 61a7f45bfad8..da1f124db69c 100644
--- a/drivers/video/backlight/lp855x_bl.c
+++ b/drivers/video/backlight/lp855x_bl.c
@@ -241,19 +241,17 @@ static int lp855x_bl_update_status(struct backlight_device *bl)
 {
 	struct lp855x *lp = bl_get_data(bl);
 	int brightness = bl->props.brightness;
-	int ret;
 
 	if (bl->props.state & (BL_CORE_SUSPENDED | BL_CORE_FBBLANK))
 		brightness = 0;
 
 	if (lp->mode == PWM_BASED)
-		ret = lp855x_pwm_ctrl(lp, brightness,
+		return lp855x_pwm_ctrl(lp, brightness,
 				      bl->props.max_brightness);
 	else if (lp->mode == REGISTER_BASED)
-		ret = lp855x_write_byte(lp, lp->cfg->reg_brightness,
+		return lp855x_write_byte(lp, lp->cfg->reg_brightness,
 					(u8)brightness);
-
-	return ret;
+	return -EINVAL;
 }
 
 static const struct backlight_ops lp855x_bl_ops = {

base-commit: 21ef7b1e17d039053edaeaf41142423810572741
-- 
2.41.0

